smart_control/dataset/dataset_test.py

```diff
@@ -0,0 +1,524 @@
+            'exhaust_fan_speed_percentage_command': 1,
+            'outside_air_damper_percentage_command': 1,
+            'supply_air_static_pressure_setpoint': 1,
+            'building_air_static_pressure_setpoint': 1,
+        },
+    }
+    self.assertEqual(device_infos[0], first_device_info)
+
+  @unittest.skipUnless(TEST_DATASET_DOWNLOAD, SKIP_REASON)
+  def test_devices_df(self):
+    devices_df = self.ds.devices_df
+    self.assertIsInstance(devices_df, pd.DataFrame)
+    self.assertEqual(len(devices_df), 173)
+
+    # each row uniquely identified by the device identifier:
+    self.assertEqual(devices_df['device_id'].nunique(), len(devices_df))
+
+    expected_column_names = [
+        'device_id',
+        'namespace',
+        'code',
+        'device_type',
+        'observable_fields',
+        'action_fields',
+    ]
+    self.assertEqual(devices_df.columns.tolist(), expected_column_names)
+
+    first_row = {
+        'device_id': '202194278473007104',
+        'namespace': 'PHRED',
+        'code': 'SB1:AHU:AC-2',
+        'device_type': 6,
+        'observable_fields': {
+            'building_air_static_pressure_sensor': 1,
+            'outside_air_flowrate_sensor': 1,
+            'supply_fan_speed_percentage_command': 1,
+            'supply_air_temperature_sensor': 1,
+            'supply_fan_speed_frequency_sensor': 1,
+            'supply_air_static_pressure_setpoint': 1,
+            'return_air_temperature_sensor': 1,
+            'mixed_air_temperature_setpoint': 1,
+            'exhaust_fan_speed_percentage_command': 1,
+            'exhaust_fan_speed_frequency_sensor': 1,
+            'outside_air_damper_percentage_command': 1,
+            'mixed_air_temperature_sensor': 1,
+            'exhaust_air_damper_percentage_command': 1,
+            'cooling_percentage_command': 1,
+            'outside_air_flowrate_setpoint': 1,
+            'supply_air_temperature_setpoint': 1,
+            'building_air_static_pressure_setpoint': 1,
+            'supply_air_static_pressure_sensor': 1,
+        },
+        'action_fields': {
+            'exhaust_air_damper_percentage_command': 1,
+            'supply_air_temperature_setpoint': 1,
+            'supply_fan_speed_percentage_command': 1,
+            'outside_air_flowrate_setpoint': 1,
+            'cooling_percentage_command': 1,
+            'mixed_air_temperature_setpoint': 1,
+            'exhaust_fan_speed_percentage_command': 1,
+            'outside_air_damper_percentage_command': 1,
+            'supply_air_static_pressure_setpoint': 1,
+            'building_air_static_pressure_setpoint': 1,
+        },
+    }
+    self.assertEqual(devices_df.iloc[0].to_dict(), first_row)
+
+  @unittest.skipUnless(TEST_DATASET_DOWNLOAD, SKIP_REASON)
+  def test_zone_infos(self):
+    zone_infos = self.ds.zone_infos
+
+    self.assertIsInstance(zone_infos, list)
+    self.assertEqual(len(zone_infos), 563)
+
+    first_zone_info = {
```

Comment:
Consider moving to a shared location like a property on the test class that can also be accessed by the df test